### PR TITLE
feat: added timeplus output

### PR DESCRIPTION
**Pull request by ye11ow**

### CODE REVIEW

File: docs/modules/components/pages/outputs/timeplus.adoc [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
++
+--
+
+```yml
+# Common config fields, showing default values
+output:
+  label: ""
+  timeplus:
+    target: timeplus
+    url: https://us.timeplus.cloud # No default (required)
```
Comment (jovezhong) [inline docs/modules/components/pages/outputs/timeplus.adoc]: us-west-2.timeplus.cloud

File: docs/modules/components/pages/outputs/timeplus.adoc [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+      check: ""
+```
+
+--
+Advanced::
++
+--
+
+```yml
+# All config fields, showing default values
```
Comment (jovezhong) [inline docs/modules/components/pages/outputs/timeplus.adoc:61]: config-> configuration

File: docs/modules/components/pages/outputs/timeplus.adoc [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+output:
+  label: ""
+  timeplus:
+    target: timeplus
+    url: https://us.timeplus.cloud # No default (required)
```
Comment (jovezhong) [inline docs/modules/components/pages/outputs/timeplus.adoc]: us-west-2

File: docs/modules/components/pages/outputs/timeplus.adoc [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+      byte_size: 0
+      period: ""
+      check: ""
+      processors: [] # No default (optional)
+```
+
+--
+======
+
+This output can send message to Timeplus Enterprise Cloud, Timeplus Enterprise Onprem or directly to timeplusd.
```
Comment (jovezhong) [inline docs/modules/components/pages/outputs/timeplus.adoc]: "Timeplus Enterprise Onprem"->"Timeplus Enterprise (self-hosted)"

File: docs/modules/components/pages/outputs/timeplus.adoc [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+
+This output accepts structured message only. It also expects all message contains the same keys and matches the schema of stream. If the upstream source returns
+unstructured message such as string, please include a processors to contruct the strcuture message. Here is a short sample:
+
+```yml
+output:
+  timeplus:
+    workspace: my_workspace_id
+    stream: mystream
+    apikey: JFmlmLDJPtrgo6cEgSDKHyg9NknZ5PM2M9lGelCwE2BtHPxlXnCbQaWua
```
Comment (jovezhong) [inline docs/modules/components/pages/outputs/timeplus.adoc]: maybe use a fake apikey, this looks too real

File: docs/modules/components/pages/outputs/timeplus.adoc [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+The url should always include schema and host. The port number should be included as well if it is not 80.
+
+
+*Type*: `string`
+
+
+```yml
+# Examples
+
+url: https://us.timeplus.cloud
```
Comment (jovezhong) [inline docs/modules/components/pages/outputs/timeplus.adoc]: us-west-2

File: internal/impl/timeplus/output.go [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+    url: http://localhost:3218
+    stream: mystream
+    username: username
+    password: pw
+` + "```" + `
+`)
+
+	outputConfigSpec.
+		Field(service.NewStringEnumField("target", http.TargetTimeplus, http.TargetTimeplusd).Default(http.TargetTimeplus)).
+		Field(service.NewURLField("url").Description("The url should always include schema and host. The port number should be included as well if it is not 80.").Examples("https://us.timeplus.cloud", "http://localhost:8000")).
```
Comment (jovezhong) [inline internal/impl/timeplus/output.go]: us-west-2

File: internal/impl/timeplus/output.go [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+		Field(service.NewStringField("workspace").Optional().Description("ID of the workspace. Required if target is `timeplus`.")).
+		Field(service.NewStringField("stream").Description("The name of the stream. Make sure the schema of the stream matches the input")).
+		Field(service.NewStringField("apikey").Secret().Optional().Description("The API key. Required if you are sending message to Timeplus Enterprise Cloud")).
+		Field(service.NewStringField("username").Optional().Description("The username. Required if you are sending message to Timeplus Enterprise Onprem or timeplusd")).
```
Comment (jovezhong) [inline internal/impl/timeplus/output.go]: Replace Onprem with self-host, to align with website/docs

File: internal/impl/timeplus/timeplus_output_test.go [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+		err = out.WriteBatch(context.Background(), batch)
+		require.NoError(t, err)
+
+		<-ch
+
+		err = out.Close(context.Background())
+		require.NoError(t, err)
+	})
+
+	t.Run("Successful send data to remote timeplus enterprise", func(t *testing.T) {
```
Comment (jovezhong) [inline internal/impl/timeplus/timeplus_output_test.go]: Prefer using `Timeplus Enterprise`

**Review by jovezhong — APPROVED** [review]

File: docs/modules/components/pages/outputs/timeplus.adoc (hunk 2 above)
Comment (ye11ow) [reply docs/modules/components/pages/outputs/timeplus.adoc:61]: this is generated by connect, it is hardcoded in benthos side

File: internal/impl/timeplus/timeplus_output_test.go (hunk 9 above)
Comment (ye11ow) [reply internal/impl/timeplus/timeplus_output_test.go]: fixed

File: internal/impl/timeplus/output.go (hunk 8 above)
Comment (ye11ow) [reply internal/impl/timeplus/output.go]: fixed

File: internal/impl/timeplus/output.go (hunk 7 above)
Comment (ye11ow) [reply internal/impl/timeplus/output.go]: fixed

File: docs/modules/components/pages/outputs/timeplus.adoc (hunk 6 above)
Comment (ye11ow) [reply docs/modules/components/pages/outputs/timeplus.adoc]: fixed

File: docs/modules/components/pages/outputs/timeplus.adoc (hunk 5 above)
Comment (ye11ow) [reply docs/modules/components/pages/outputs/timeplus.adoc]: fixed

File: docs/modules/components/pages/outputs/timeplus.adoc (hunk 4 above)
Comment (ye11ow) [reply docs/modules/components/pages/outputs/timeplus.adoc]: fixed

File: docs/modules/components/pages/outputs/timeplus.adoc (hunk 3 above)
Comment (ye11ow) [reply docs/modules/components/pages/outputs/timeplus.adoc]: fixed

File: docs/modules/components/pages/outputs/timeplus.adoc (hunk 1 above)
Comment (ye11ow) [reply docs/modules/components/pages/outputs/timeplus.adoc]: fixed

**Pull request closed without merging**